main.py

```diff
@@ -0,0 +1,227 @@
+import os
+import time
+from collections import deque
+from cProfile import run
+
+import numpy as np
+import torch
+
+from agent import algo, utils
+from agent.arguments import get_args
+from agent.envs import make_vec_envs
+from agent.policy import Policy
+from agent.storage import RolloutStorage
+from evaluation import evaluate
+from logger import get_logger
+
+
+def main():
+    args = get_args()
+    
+    if args.log:
+        params, logger = get_logger()
+    else:
+        params, logger = None, None
+
+    torch.manual_seed(args.seed)
+    torch.cuda.manual_seed_all(args.seed)
+
+    if args.cuda and torch.cuda.is_available() and args.cuda_deterministic:
+        torch.backends.cudnn.benchmark = False
+        torch.backends.cudnn.deterministic = True
+
+    log_dir = os.path.expanduser(args.log_dir)
+    eval_log_dir = log_dir + "_eval"
+    utils.cleanup_log_dir(log_dir)
+    utils.cleanup_log_dir(eval_log_dir)
+
+    torch.set_num_threads(1)
+    device = torch.device("cuda:0" if args.cuda else "cpu")
+
+    envs = make_vec_envs(
```

Comment:
It looks like you are not yet initializing `ASTEnv`. When and wherever you do, make sure to wrap it in a [`TimeLimit`](https://github.com/openai/gym/blob/master/gym/wrappers/time_limit.py) wrapper.

Reply:
I've added this wrapper, but how many `max_episode_steps` do we usually need?

Reply:
As I said in our meeting, that should be a parameter, not a magic number.